### PR TITLE
Add support for vDPA test environment

### PR DESCRIPTION
**Pull request by yanglei-rh**
Add support for vDPA test environment.

ID:2035984
Signed-off-by: Lei Yang <leiyang@redhat.com>

### CODE REVIEW

**Comment by yanglei-rh** [comment]
Hello @luckyh @PaulYuuu @YongxueHong Could you please help me review this patch when you have free time? Many thanks.

File: virttest/utils_vdpa.py [hunk 1]
```diff
@@ -261,3 +266,23 @@ def get_vdpa_pci(driver='mlx5_core'):
     for pci_info in pf_info.values():
         if pci_info.get("driver", "") == driver:
             return pci_info.get('pci_id')
+
+
+def get_dev_file_by_name(name):
```
Comment (YongxueHong) [inline virttest/utils_vdpa.py:271]: Suggest adding vdpa string to name it.
```suggestion
def get_vdpa_path_by_name(name):
```

Comment (yanglei-rh) [reply virttest/utils_vdpa.py]: Support your point of view, it has been changed, thank you very much.

**Review by YongxueHong — COMMENTED** [review]
LGTM

**Review by YongxueHong — APPROVED** [review]

**Comment by yanglei-rh** [comment]
Hello @luckyh @PaulYuuu Could you please help me review it? Thanks in advance.

**Comment by PaulYuuu** [comment]
Here are the generated command lines
```
[stdlog]     -device pcie-root-port,id=pcie-root-port-4,port=0x4,addr=0x1.0x4,bus=pcie.0,chassis=5 \
[stdlog]     -device virtio-net-pci,mac=f0:11:22:33:44:00,rombar=0,id=id7hygP0,netdev=idVxL5qr,bus=pcie-root-port-4,addr=0x0  \
[stdlog]     -netdev vhost-vdpa,id=idVxL5qr,vhostdev=/dev/vhost-vdpa-0  \
```
```
# grep "f0:11:22:33:44:00" ./ip-sniffer.log
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 0.0.0.0 255.255.255.255 1 0x991a2b4d f0:11:22:33:44:00 0.0.0.0 3
2022-11-10 03:49:03: 10.16.203.253 10.16.203.190 2 0x991a2b4d f0:11:22:33:44:00 10.16.203.190 5
```

But sometimes the guest cannot get an IP address, seems the vDPA performance issue, can ignore it this time.

**Review by luckyh — APPROVED** [review]

**Comment by luckyh** [comment]
Thanks to all.

**Pull request merged**